src/ci/run.sh

```diff
@@ -8,7 +8,7 @@ fi
 
 if [ "$NO_CHANGE_USER" = "" ]; then
   if [ "$LOCAL_USER_ID" != "" ]; then
-    useradd --shell /bin/bash -u $LOCAL_USER_ID -o -c "" -m user
+    id -u $LOCAL_USER_ID &>/dev/null || useradd --shell /bin/bash -u $LOCAL_USER_ID -o -c "" -m user
```

Comment:
This change doesn't seem like it should be in this PR

Reply:
Thanks for the feedback! I'm not quite sure whether you're asking for context, or telling me to revert this change? 

I remember this change being intentional as part of [605858b](https://github.com/rust-lang/rust/commit/605858b83f8c4cfc889afca0374ac4c078033f5e), which I believe I could most effectively test using the CI infrastructure's docker containers for acquiring a ubuntu-flavored LLVM 14. IIRC every part of the CI process was idempotent/well-suited to this purpose except the `useradd` here. That would fail to create a user that already exists, and it seemed a shame to me to throw away the whole container's context because of one extra entry in `/etc/passwd`.

I'm happy to work with you on making sure the change is appropriate here (I love me a good dense shellism, but I understand why not everyone does :sweat_smile:), or to move it to a different PR if you think that'd be a better home for it. Just let me know!

Reply:
Another PR is better